packages/shared/toShallowRef/index.ts

```diff
@@ -0,0 +1,21 @@
+import type { ComputedRef, Ref, ShallowRef } from 'vue-demi'
+import {
+  isRef,
+  shallowRef,
+} from 'vue-demi'
+import type { MaybeRefOrGetter } from '../utils'
+import { toRef } from '../toRef'
+
+/**
+ * Normalize value/ref/getter to `shallowRef` or `computed`.
+ */
+export function toShallowRef<T>(r: () => T): Readonly<Ref<T>>
+export function toShallowRef<T>(r: ComputedRef<T>): ComputedRef<T>
+export function toShallowRef<T>(r: Ref<T>): Ref<T>
+export function toShallowRef<T>(r: MaybeRefOrGetter<T>): Ref<T>
+export function toShallowRef<T>(r: T): ShallowRef<T>
+export function toShallowRef<T>(r: MaybeRefOrGetter<T>) {
+  return (isRef(r) || typeof r === 'function')
+    ? toRef(r)
```

Comment:
I am not sure if this could be potentially confusing.

Say if I have:

```ts
const foo = ref({ nested: 1 })
const bar = toShallowRef(foo)

// bar is actually deep ref not shallow
```

Semantically I feel that users might also expect `toShallowRef` to convert a normal ref to shallow ref.

Reply:
Sorry for the delay.

I agree. I could certainly see it being counterintuitive. It seems as though the behavior of `ref()` and `shallowRef()` also [employs the same logic](https://github.com/vuejs/core/blob/28a6f05bf52f95a7cc06acca8205ec386f174601/packages/reactivity/src/ref.ts#L128C10-L128C10), however. I'm assuming one would have to dig further into raw values or clone the object in order to strip reactivity proxies. **I think it makes sense to defer to Vue core behavior here.**

If the above makes sense, do you think it's worthwhile to put this in the documentation page as a potential "gotcha" of the function?